parsing/extensions.mli

```diff
@@ -97,7 +97,12 @@ module type AST = sig
       if it's not a language extension term, return [None]; if it's a disabled
       language extension term, raise an error.
 
-      AN IMPORTANT NOTE: We indent calls to this function *very* strangely: we
+      AN IMPORTANT NOTE: The design of this function is careful to make
+      merge conflicts with upstream less likely: we want no edits at all
+      to surrounding code. This is why we return a [t option], not some
+      structure that could include the [ast_desc] if there is no extension.
+
+      In addition, we indent calls to this function *very* strangely: we
```

Comment:
I'd like to prioritize the note about indentation, since that's the first thing *users* of the function will note.  I tried moving your documentation to the bottom and putting it in its own paragraph; see the other suggestion.

```suggestion
      AN IMPORTANT NOTE: We indent calls to this function *very* strangely: we
```

Reply:
I'm going to mark this as resolved, but we can keep iterating in another PR if there are further improvements.